Add SPEEDYBEEH5DEV

haslinghuis opened the pull request:
This target is still in development so does not compile currently.
- note CC and SERVO are mutually exclusive

IDX | PIN | RESOURCE | TIMER | DMA
---: | --- | --- | --- | ---:
0 | PB6 | MOTOR1_PIN | TIM4_CH1 | 0 
1 | PB7 | MOTOR2_PIN | TIM4_CH2 | 0
2 | PB0 | MOTOR3_PIN | TIM3_CH3 | 0
3 | PB1 | MOTOR4_PIN | TIM3_CH4 | 0
4 | PC9 | MOTOR5_PIN | TIM8_CH4 | 0
5 | PC8 | MOTOR6_PIN | TIM8_CH3 | 0
6 | PB11 | MOTOR7_PIN | TIM2_CH4 | 0
7 | PB10 | MOTOR8_PIN | TIM2_CH3 | 0
8 | PA8 | LED_STRIP_PIN | TIM1_CH1 | 0
9 | PA0 | CAMERA_CONTROL_PIN | TIM5_CH1 | -1

Timer specs: https://github.com/betaflight/betaflight/pull/13685

### Status

MCU: STM32H563xx CLK=250MHz, Vref=3.26V, Core temp=37degC
STACK: 2048b (0x2009fff8)
CONFIG: CONFIGURED (4259b / 131072b)

DEVICES DETECTED: SPI=1, I2C=1 (0 errors)
GYRO: (1) LSM6DSV16X enabled locked dma
ACC: LSM6DSV16X
BARO: LPS22DF
GPS: NOT ENABLED
OSD: MSP (53 x 20)
SD-CARD: Manufacturer 0x3, 31166976kB, 01/2022, v8.5, 'SD32G FS: Ready

System Uptime: 38 seconds, Current Time: 2026-06-03T12:27:09.474+00:00
CPU:49%, cycle time: 135, GYRO rate: 7407, RX rate: 0, System rate: 9
Voltage: 0.00V (0S battery - NOT PRESENT)
Arming disable flags: RXLOSS ANGLE CLI MSP

### Support ID

`d65abae7-dbd9-478b-b8a6-41583778f0bc`

---

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added support for the under-development SPEEDYBEEH5DEV STM32H5 flight controller target.
  * Enabled onboard accelerometer, gyroscope, barometer, video overlay, SD card, motor, LED, camera, and telemetry capabilities.
  * Added configured serial, SPI, I2C, ADC, and sensor interfaces for the board.
  * Added default blackbox logging and voltage/current monitoring support.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

haslinghuis committed:
Add SPEEDYBEEH5

haslinghuis committed:
Add timer map

haslinghuis committed:
Update timer map

haslinghuis committed:
Update config.h

haslinghuis committed:
Update TIMER_PIN_MAPPING and ADC_DMA options

kedeng commented on configs/SPEEDYBEEH5DEV/config.h:
```diff
@@ -0,0 +1,144 @@
+#define I2C1_SCL_PIN                    PB8
+#define I2C1_SDA_PIN                    PB9
+
+#define LED0_PIN                        PC3
```
PC3  ->  LED0_PIN
SPI2_MOSI -> PB15

kedeng commented on configs/SPEEDYBEEH5/config.h:
```diff
@@ -0,0 +1,144 @@
+#define SPI1_SDO_PIN                    PA7
+
+#define SPI2_SCK_PIN                    PB13
+#define SPI2_SDI_PIN                    PC2
```
PC2 -> RSSI_ADC
SPI2_SDI_PIN -> PB14

kedeng commented on configs/SPEEDYBEEH5/config.h:
```diff
@@ -0,0 +1,144 @@
+#define SPI1_SDO_PIN                    PA7
+
+#define SPI2_SCK_PIN                    PB13
+#define SPI2_SDI_PIN                    PC2
```
#define SPI2_SDI_PIN   PB14   // MISO
#define SPI2_SDO_PIN   PB15   // MOSI

haslinghuis committed:
Fix SPI2 and remove S10

Updated SPI2 pins to use PB14 and PB15 instead of PC2 and PC3.

haslinghuis committed:
Update config.h by removing SERVO1_PIN

Removed SERVO1_PIN from configuration and updated TIMER_PIN_MAPPING to exclude it.

ot0tot reviewed (approved):
Approving for use as dev board as requested

haslinghuis committed:
Change board name to SPEEDYBEEH5DEV
